app/_src/gateway/production/debug-request.md

```diff
@@ -0,0 +1,162 @@
+
+* `rewrite`
+* `access`
+* `balancer`
+* `response`
+* `header_filter`
+* `body_filter`
+* `upstream`
+* `log`
+
+If this header isn't present or contains an unknown value, timing information will not be collected for the current request.
+
+### X-Kong-Request-Debug-Log header
+
+If the `X-Kong-Request-Debug-Token` header is set to true, timing information will also be logged in the {{site.base_gateway}} error log with a log level of notice. By default, the `X-Kong-Request-Debug-Token` header is set to `false`. The log line will have the `[request-debug]` prefix to aid in searching.
+
+## Debug request example 
+
+The following is an example debug request:
+
+```bash
+curl http://kong_proxy_host:8000/example \
+    -H "X-Kong-Request-Debug: *" \
+    -H "X-Kong-Request-Debug-Token: xxxxxx"
+```
+
+Here's an example of the output of the response header:
+
+```bash
+{
+    "request_id": "2effa21fb2d36d31f80ed02635cde86b",
+    "workspace_id": "7b7f79f2-8d52-470c-a307-e76f986041cf",
+    "child": {
+        "rewrite": {
+            "total_time": 0
+        },
+        "access": {
+            "child": {
+                "dns": {
+                    "child": {
+                        "example.host": {
+                            "child": {
+                                "resolve": {
+                                    "total_time": 0,
+                                    "cache_hit": true
+                                }
+                            },
+                            "total_time": 0
+                        }
+                    },
+                    "total_time": 0
+                }
+            },
+            "total_time": 1
+        },
+        "header_filter": {
+            "total_time": 0
+        },
+        "balancer": {
+            "total_time": 0
+        },
+        "upstream": {
+            "total_time": 100,
+            "child": {
+                "time_to_first_byte": {
+                    "total_time": 20
+                },
+                "streaming": {
+                    "Total_time": 80
+                }
+            }
+        }
+    }
+}
+```
+
+You can analyze the debug output like the following:
+* The unit of `total_time` is `millisecond`.
+* The DNS resolution for `example.host` was cached, which is why it's so fast in the example.
+* Upstream took 100ms in this request.
+    * The elapsed time from {{site.base_gateway}} sending the request to the upstream to {{site.base_gateway}} receiving the first byte is 20ms.
+    * The elapsed time from the first byte to the last byte from the upstream is 80ms.
+
+You can also filter the debug output:
+
+```bash
+curl http://kong_proxy_host:8000/example \
+    -H "X-Kong-Request-Debug: upstream" \
+    -H "X-Kong-Request-Debug-Token: xxxxxx"
+```
+
+This will return something like the following:
+
+```bash
+X-Kong-Request-Id: a1a1530f8ddb6f6f2462916ae002b715
```

Comment:
❗ For reviewers: The example output for this example and the following ones didn't match the previous one in the draft doc. I left them as-is just to get this out the door quickly. Should we make them match the first one or is this okay?